packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewOutputList/TransactionReviewOutput.tsx

```diff
@@ -32,31 +36,88 @@ const getFeeLabel = (networkType: NetworkType) => {
     }
 };
 
-const getDisplayModeStringsMap = (
+const getTranslationValues = (
     isEverstake: boolean,
-): Record<StakeType, { value: TranslationKey; label: TranslationKey }> => ({
-    stake: {
-        value: 'TR_STAKE_ON_EVERSTAKE',
-        label: 'TR_STAKE_STAKE',
-    },
-    unstake: {
-        value: isEverstake ? 'TR_UNSTAKE_FROM_EVERSTAKE' : 'TR_UNSTAKE_FROM_STAKE_ACCOUNT',
-        label: 'TR_STAKE_UNSTAKE',
-    },
-    claim: {
-        value: isEverstake ? 'TR_CLAIM_FROM_EVERSTAKE' : 'TR_CLAIM_FROM_STAKE_ACCOUNT',
-        label: 'TR_STAKE_CLAIM',
-    },
-});
+    stakeType?: StakeType,
+    evmApprovalType?: Exclude<EvmTransactionPurpose, 'default'>,
+    device?: TrezorDevice,
+): Record<'value' | 'label', TranslationKey> | null => {
+    const firmwareVersion = getFirmwareVersion(device);
+    const isNewApproveFlow = firmwareVersion && isNewerOrEqual(firmwareVersion, '2.9.0');
+
+    const approvalStrings: Record<
+        Exclude<EvmTransactionPurpose, 'default'>,
+        Record<'value' | 'label', TranslationKey>
+    > = {
+        approval: {
+            value: 'TR_APPROVE_DESCRIPTION',
+            label: 'TR_APPROVE_TITLE',
+        },
+        revoke: {
+            value: 'TR_REVOKE_DESCRIPTION',
+            label: 'TR_REVOKE_TITLE',
+        },
+    };
+
+    if (evmApprovalType && !isNewApproveFlow) {
+        return null;
+    }
+
+    if (evmApprovalType) {
+        return approvalStrings[evmApprovalType];
+    }
+
+    if (stakeType) {
+        const stakeStrings: Record<StakeType, Record<'value' | 'label', TranslationKey>> = {
+            stake: {
+                value: 'TR_STAKE_ON_EVERSTAKE',
+                label: 'TR_STAKE_STAKE',
+            },
+            unstake: {
+                value: isEverstake ? 'TR_UNSTAKE_FROM_EVERSTAKE' : 'TR_UNSTAKE_FROM_STAKE_ACCOUNT',
+                label: 'TR_STAKE_UNSTAKE',
+            },
+            claim: {
+                value: isEverstake ? 'TR_CLAIM_FROM_EVERSTAKE' : 'TR_CLAIM_FROM_STAKE_ACCOUNT',
+                label: 'TR_STAKE_CLAIM',
+            },
+        };
+
+        return stakeStrings[stakeType];
+    }
+
+    return null;
+};
+
+const getContractTitle = (
+    isSolana: boolean,
```

Comment:
Can you please change this `isSolana`? would be better to send networkType and resolve it here